targets/TARGET_STM/qspi_api.c

```diff
@@ -249,10 +278,40 @@ void qspi_prepare_command(const qspi_command_t *command, QSPI_CommandTypeDef *st
         st_command->AlternateByteMode = QSPI_ALTERNATE_BYTES_NONE;
         st_command->AlternateBytesSize = 0;
     } else {
-        st_command->AlternateBytes = command->alt.value;
+        // Alt size must be a multiple of the number of bus lines used (i.e. a whole number of cycles)
+        if ((alt_lines == 0) || (command->alt.size % alt_lines != 0)) {
+            return QSPI_STATUS_ERROR;
+        }
+
+        // Round up to nearest byte - unused parts of byte act as dummy cycles
+        uint32_t rounded_size = ((command->alt.size - 1) >> 3) + 1;
+        // Maximum of 4 alt bytes
+        if (rounded_size > 4) {
+            return QSPI_STATUS_ERROR;
+        }
+        
+        // Unused bits in most significant byte of alt
+        uint8_t leftover_bits = (rounded_size << 3) - command->alt.size;
+        if (leftover_bits != 0) {
+            // Account for dummy cycles that will be spent in the alt portion of the command
+            uint8_t integrated_dummy_cycles = leftover_bits / alt_lines;
+            if (st_command->DummyCycles < integrated_dummy_cycles)
+            {
+                // Not enough dummy cycles to account for a short alt
+                return QSPI_STATUS_ERROR;
+            }
+            st_command->DummyCycles -= integrated_dummy_cycles;
+
+            // Align alt value to the end of the most significant byte
+            st_command->AlternateBytes = command->alt.value << leftover_bits;
+        } else {
+            rounded_size -= 1;
```

Comment:
I think this line shouldn't be here. It cause that `rounded_size` falls below 0 in some cases

Reply:
I agree. Addressed in 9b32c0f pushed to #11602 , and rebased this PR on #11602 again